src/transport/SessionManager.cpp

```diff
@@ -555,6 +551,10 @@ void SessionManager::SecureUnicastMessageDispatch(const PacketHeader & packetHea
     err = secureSession->GetSessionMessageCounter().GetPeerMessageCounter().Verify(packetHeader.GetMessageCounter());
     if (err == CHIP_ERROR_DUPLICATE_MESSAGE_RECEIVED)
     {
+        ChipLogDetail(Inet,
+                      "Received a duplicate message with MessageCounter:" ChipLogFormatMessageCounter
+                      " on exchange " ChipLogFormatExchangeId,
+                      packetHeader.GetMessageCounter(), ChipLogValueExchangeIdFromReceivedHeader(payloadHeader));
         isDuplicate = SessionMessageDelegate::DuplicateMessage::Yes;
```

Comment:
why is `isDuplicate` an enum class with Yes/No. Why not a bool?

Reply:
Because at callsites a bare bool is not very clear as to what it means, so enums are used, as in this case, for clarity.

Reply:
I agree that enum is better than bool. From call site, passing a SessionMessageDelegate::DuplicateMessage::Yes/No is more intuitive than true/false.

Reply:
Anyway it should be another PR and should not block this PR